systemtest/src/test/java/io/strimzi/systemtest/report/ClusterReportST.java

```diff
@@ -0,0 +1,255 @@
+            "--namespace=" + testStorage.getNamespaceName(),
+            "--cluster=non-existing-k"
+        ));
+    }
+
+    @ParallelTest
+    void createReportWithUnknownSecretVerbosity(TestInfo testInfo) {
+        LOGGER.info("Running report on {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        assertThrows(KubeClusterException.class, () ->
+            Exec.exec(USER_PATH + "/../tools/report.sh",
+                "--namespace=" + testStorage.getNamespaceName(),
+                "--cluster=" + testStorage.getClusterName(),
+                "--secrets=unknown"
+            ));
+    }
+
+    @BeforeAll
+    void setUp(final ExtensionContext extensionContext) {
+        this.clusterOperator = this.clusterOperator
+            .defaultInstallation(extensionContext)
+            .withReplicas(2)
+            .createInstallation()
+            .runInstallation();
+        this.testStorage = new TestStorage(extensionContext);
+        resourceManager.createResourceWithWait(extensionContext,
+            KafkaTemplates.kafkaEphemeral(testStorage.getClusterName(), 3, 3).build(),
+            KafkaTemplates.kafkaEphemeral(testStorage.getClusterName() + "-tgt", 3).build()
+        );
+        resourceManager.createResourceWithWait(extensionContext,
+            KafkaBridgeTemplates.kafkaBridge(BRIDGE_NAME,
+                KafkaResources.plainBootstrapAddress(testStorage.getClusterName()), 1).build());
+        resourceManager.createResourceWithWait(extensionContext,
+            KafkaConnectTemplates.kafkaConnect(
+                CONNECT_NAME, testStorage.getNamespaceName(), testStorage.getClusterName(), 1).build());
+        resourceManager.createResourceWithWait(extensionContext,
+            KafkaMirrorMaker2Templates.kafkaMirrorMaker2(
+                MM2_NAME, testStorage.getClusterName() + "-tgt", testStorage.getClusterName(), 1, false).build());
+    }
+
+    private String buildOutPath(TestInfo testInfo, String clusterName) {
+        String methodName = testInfo.getTestMethod().isPresent() ?
+            testInfo.getTestMethod().get().getName() : UUID.randomUUID().toString();
+        return USER_PATH + "/target/reports/" + clusterName + "/" + methodName;
+    }
+
+    private void assertValidReport(String outPath) {
+        assertThat("Output directory does not exist", Files.exists(Path.of(outPath)));
+        assertThat("Output ZIP does not exist",
+            FileUtils.listFilesWithSuffix(outPath, ".zip").length == 1);
+        assertThat("Kafka CRD does not exist",
+            Files.exists(Path.of(outPath + "/reports/crds/kafkas.kafka.strimzi.io.yaml")));
+        assertThat("ClusterOperator logs are missing", FileUtils.listFilesWithPrefix(
+            outPath + "/reports/podlogs", "strimzi-cluster-operator").length == 2);
+        assertThat("Kafka logs are missing", FileUtils.listFilesWithPrefix(
+            outPath + "/reports/podlogs", testStorage.getClusterName() + "-kafka").length == 3);
```

Comment:
my concern is that we are validating just a few of the things that the report is returning, or? What about all the other stuff which are not pod logs?

Reply:
Yes, I'll add more checks. If you have any suggestion, feel free to raise.

Reply:
It's bad to say but maybe testing all files it can produce (based on the tests we have here)? I know it could be a lot ...